lern-data-products/src/main/scala/org/sunbird/core/util/DataSecurityUtil.scala

```diff
@@ -28,23 +28,55 @@ object DataSecurityUtil {
    */
   def getSecurityLevel(jobId: String, orgId: String): String = {
     JobLogger.log(s"getSecurityLevel jobID:: $jobId orgid:: $orgId", None, INFO)(new String())
-    val requestBody = Map("request" -> Map("orgId" -> orgId, "key" -> "dataSecurityPolicy"))
+    val httpResponseBody = getTenantPreferanceDetails(orgId, "dataSecurityPolicy")
+    val responseBody = JSONUtils.deserialize[Map[String, AnyRef]](httpResponseBody)
+    val data = responseBody.getOrElse("result", Map[String, AnyRef]()).asInstanceOf[Map[String, AnyRef]]
+      .getOrElse("response", Map[String, AnyRef]()).asInstanceOf[Map[String, AnyRef]]
+      .getOrElse("data", Map[String, AnyRef]()).asInstanceOf[Map[String, AnyRef]]
+    val globalLevel = data.getOrElse("level", "").asInstanceOf[String]
+    val jobDetail = data.getOrElse("job", Map[String, AnyRef]()).asInstanceOf[Map[String, AnyRef]]
+      .getOrElse(jobId, Map[String, AnyRef]()).asInstanceOf[Map[String, AnyRef]]
+    val jobLevel = jobDetail.getOrElse("level", "").asInstanceOf[String]
+    if (!StringUtils.isEmpty(jobLevel)) jobLevel else globalLevel
+  }
+
+  /**
+   * fetch the PII fields by calling tenant preference read API using orgId
+   *
+   * @param jobId
+   * @param orgId
+   * @return
+   */
+  def getPIIFieldDetails(jobId: String, orgId: String): List[String] = {
+    JobLogger.log(s"getSecurityLevel jobID:: $jobId orgid:: $orgId", None, INFO)(new String())
+    val httpResponseBody = getTenantPreferanceDetails(orgId, "userPrivateFields")
+    val responseBody = JSONUtils.deserialize[Map[String, AnyRef]](httpResponseBody)
+    val data = responseBody.getOrElse("result", Map[String, AnyRef]()).asInstanceOf[Map[String, AnyRef]]
+      .getOrElse("response", Map[String, AnyRef]()).asInstanceOf[Map[String, AnyRef]]
+      .getOrElse("data", Map[String, AnyRef]()).asInstanceOf[Map[String, AnyRef]]
+    val piiFields = data.getOrElse("PIIFields", List[String]()).asInstanceOf[List[String]]
```

Comment:
Please change `PIIFields` to camel case `piiFields`. Please update the same in TenantPreference request body in microsite as well.